### ISSUE TITLE
Anonymous Authentication in any case

### ISSUE THREAD

**Issue by koromkorom** [cotect/cotect#11]
**Describe the bug:**

Any time a user signs up with a phone number, a firebase user is created. After clicking the "Next" button, also an anonymous user is created. 

**Expected behaviour:**

User gives phone number -> Phone Number sign-in
User skips phone number -> Anonymous sign-in

**Steps to reproduce the issue:**

Just sign in with phone number. 

**Technical details:**

Developer branch

**Possible Fix:**

The problem is, that with click on next, the onSignInAnonymously function is called:

https://github.com/cotect/cotect/blob/81bd75b1e9f0745cdbf583301095096b084b7665/app/mobile/screens/steps/PhoneNumberStep.js#L49-L62

In this function `auth().signInAnonymously();` is called. 

A fix would be to check if  a user is already signed in like this: 
```js
if (auth().currentUser === null) {
    try { 
        await auth().signInAnonymously(); 
    ...
}
```

**Sth to think about**

Maybe this was the origin of this problem: 

I don't see a way with firebase to "sign-in" users with their phone numbers without firebase saving their numbers online. 

![image](https://user-images.githubusercontent.com/46714635/77367725-0f0a3480-6d5b-11ea-9540-c89fafc19ec8.png)

Although this is completely separate from the database, it is not really what the app is promising. The Phone Number sign-in feature works well, but to send an sms anywhere you need to have the recipients number.

**Comment by raethlein** [comment]
Hey, thanks for reporting this! I think you are right and applied the proposed fix. I checked the behavior in the Firebase console and now either a signed-in user is stored or an anonymous user.

I will follow up on the saved number in Firebase and the mentioned issue with that conflicting with the app's promise.

**Comment by lukasmasuch** [comment]
@koromkorom Thanks for the issue report! From a technical perspective we do not need the user or the phone property within Firebase. We want to use phone auth for verification and not really for authentication. So, after the initial verification via phone, we might just delete the phone property or user from Firebase. However, this also requires some small changes within our backend.

**Comment by koromkorom** [comment]
I think immediately deleting the user might be something worth trying. However if this is for validation only, why using anonymous sign in at all?

**Comment by lukasmasuch** [comment]
In our current version, the user can skip the verification. Thereby, the report data will be marked as not-verified in our database. For this, we use the anonymous sign-in from firebase which allows us to keep a "user session" as long as the user does not clear the app data. This enables an anonymous user to update his report, get assessments, or delete his data at any future point (in case he does not delete app data).

**Comment by raethlein** [comment]
Can be closed due to changes made by #15
